pkg/aggregator/check_sampler.go

```diff
@@ -96,6 +96,11 @@ func (cs *CheckSampler) addBucket(bucket *metrics.HistogramBucket) {
 		}
 		cs.lastBucketValue[contextKey] = rawValue
 		cs.lastSeenBucket[contextKey] = time.Now()
+
+		// Return early so we don't report the first raw value instead of the delta which will cause spikes
+		if !bucketFound {
+			return
+		}
```

Comment:
let's refactor this a little with the earlier `if` statement above (let's change that statement to `if !bucketFound` and return early there).

Also, in the earlier `if` statement above, we end end up setting `cs.lastSeenBucket[contextKey] = time.Now()` twice, which is useless, so we can fix that while we're at it